src/main/java/io/github/cottonmc/libcd/api/AdvancementRewardsManager.java

```diff
@@ -0,0 +1,26 @@
+package io.github.cottonmc.libcd.api;
+
+import com.google.common.collect.Maps;
+import com.google.gson.JsonDeserializer;
+import net.minecraft.server.network.ServerPlayerEntity;
+import net.minecraft.util.Identifier;
+
+import java.util.Map;
+import java.util.function.BiConsumer;
+import java.util.function.Consumer;
+
+public class AdvancementRewardsManager {
+    public static final Map<Identifier, JsonDeserializer<Object>> DESERIALIZERS = Maps.newHashMap();
```

Comment:
It almost seems like it'd be better to just pass the implementer the `JsonObject` and have them do what they want with it. Not 100% sure at the moment.